src/lib.rs

```diff
@@ -132,23 +132,15 @@
 //! * structure!() macro takes a literal string as an argument.
 //! * It's called `structure` because `struct` is a reserved keyword in Rust.
 
-#[macro_use]
-extern crate proc_macro_hack;
 
 #[doc(hidden)]
-pub extern crate byteorder;
+pub extern crate core2;
 
+#[doc(hidden)]
+pub use core2::io::{Cursor, Error, ErrorKind, Read, Result, Write};
```

Comment:
IMO this should only be used from `core2` if `std` is not active. Otherwise import from `std::io`

Reply:
So, I've since squashed this PR into two commits, but it seems this review is out of date, as on my branch I can't see this line. I've since changed it to only use `core2` if `std` isn't enabled, as-per your review.